src/containers/TaskRuns/TaskRuns.js

```diff
@@ -0,0 +1,213 @@
+    loading: true,
+    selectedStepId: null,
+    selectedTaskId: null,
+    taskRuns: []
+  };
+
+  async componentDidMount() {
+    try {
+      const { match } = this.props;
+      const { taskName } = match.params;
+      await this.loadTaskRuns(taskName);
+    } catch (error) {
+      let message = error;
+      if (error.response) {
+        message = error.response.status === 404 ? 'Not Found' : 'Error';
+      }
+      this.setState({
+        notification: { kind: 'error', message },
+        loading: false
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { match } = this.props;
+    const { taskName } = match.params;
+    if (taskName !== prevProps.match.params.taskName) {
+      this.loadTaskRuns(taskName);
+    }
+  }
+
+  handleTaskSelected = (selectedTaskId, selectedStepId) => {
+    this.setState({ selectedStepId, selectedTaskId });
+  };
+
+  async loadTaskRuns(selectedTaskName) {
+    let notification;
+    const tasks = await getTasks();
+    const task = tasks.find(
+      currentTask => currentTask.metadata.name === selectedTaskName
+    );
+    let taskRuns = await getTaskRuns();
+    taskRuns = taskRuns
+      .filter(
+        taskRun =>
+          taskRun.spec.taskRef && taskRun.spec.taskRef.name === selectedTaskName
+      )
+      .map(taskRun => {
+        const taskName = taskRun.spec.taskRef.name;
+        const taskRunName = taskRun.metadata.name;
+        const { reason, status: succeeded } = getStatus(taskRun);
+        const pipelineTaskName = taskRunName;
+        const runSteps = this.steps(task, taskRun.status.steps);
+        const { startTime } = taskRun.status;
+        return {
+          id: taskRun.metadata.uid,
+          pipelineTaskName,
+          pod: taskRun.status.podName,
+          reason,
+          steps: runSteps,
+          succeeded,
+          taskName,
+          taskRunName,
+          startTime
+        };
+      });
+
+    if (taskRuns.length === 0) {
+      notification = {
+        kind: 'info',
+        message: 'Task has never run'
+      };
+    }
+
+    this.setState({ taskRuns, notification, loading: false });
+  }
+
+  steps(task, stepsStatus) {
+    const steps = task.spec.steps.map((step, index) => {
+      const stepStatus = stepsStatus ? stepsStatus[index] : {};
+      let status;
+      let reason;
+      if (stepStatus.terminated) {
+        status = 'terminated';
+        ({ reason } = stepStatus.terminated);
+      } else if (stepStatus.running) {
+        status = 'running';
+      } else if (stepStatus.waiting) {
+        status = 'waiting';
+      }
+
+      return {
+        ...step,
+        reason,
+        status,
+        stepStatus,
+        stepName: step.name,
+        id: step.name
+      };
+    });
+    return steps;
+  }
+
+  // once redux store is available errors will be handled properly with dedicated components
+  notification(notification) {
+    const { kind, message } = notification;
+    const titles = {};
```

Comment:
```
const titles = {
  info: '...',
  error: '...'
};
```

TODO: 
Also might re-word the info text. 'Task runs not available' could sound like an error.
There's a similar message above: https://github.com/tektoncd/dashboard/pull/49/files#diff-ccd95d56ab624312e691721ff80a1260R101 'Task has never run'.

Let's revisit this as part of the design update.